pkg/controllers/node/termination/controller.go

```diff
@@ -94,114 +95,243 @@ func (c *Controller) finalize(ctx context.Context, node *corev1.Node) (reconcile
+		// case we should set the status condition to false (requeing if it wasn't already) and then fall through to instance
+		// termination.
+		stored := nodeClaim.DeepCopy()
+		if modified := nodeClaim.StatusConditions().SetFalse(v1.ConditionTypeVolumesDetached, "TerminationGracePeriodElapsed", "TerminationGracePeriodElapsed"); modified {
+			if err := c.kubeClient.Status().Patch(ctx, nodeClaim, client.MergeFromWithOptions(stored, client.MergeFromWithOptimisticLock{})); err != nil {
+				if errors.IsConflict(err) {
+					return &reconcile.Result{Requeue: true}, nil
+				}
+				if errors.IsNotFound(err) {
+					return &reconcile.Result{}, c.associatedNodeClaimError(err)
+				}
+				return &reconcile.Result{}, err
+			}
+			// We requeue after a patch operation since we want to ensure we read our own writes before any subsequent
+			// operations on the NodeClaim.
+			return &reconcile.Result{RequeueAfter: 1 * time.Second}, nil
 		}
 	}
+	return nil, nil
+}
+
+// awaitInstanceTermination will initiate instance termination and continue to requeue until the cloudprovider indicates
+// the instance is no longer found. Once gone, the node's finalizer will be removed, unblocking the NodeClaim lifecycle
+// controller.
+func (c *Controller) awaitInstanceTermination(
+	ctx context.Context,
+	nodeClaim *v1.NodeClaim,
+	node *corev1.Node,
+	_ *time.Time,
+) (*reconcile.Result, error) {
+	isInstanceTerminated, err := termination.EnsureTerminated(ctx, c.kubeClient, nodeClaim, c.cloudProvider)
+	if client.IgnoreNotFound(err) != nil {
+		// 409 - The nodeClaim exists, but its status has already been modified
+		if errors.IsConflict(err) {
+			return &reconcile.Result{Requeue: true}, nil
+		}
+		return &reconcile.Result{}, fmt.Errorf("ensuring instance termination, %w", err)
+	}
+	if !isInstanceTerminated {
+		return &reconcile.Result{RequeueAfter: 5 * time.Second}, nil
+	}
 	if err := c.removeFinalizer(ctx, node); err != nil {
-		return reconcile.Result{}, err
+		return &reconcile.Result{}, err
 	}
-	return reconcile.Result{}, nil
+	return nil, nil
 }
 
-func (c *Controller) deleteAllNodeClaims(ctx context.Context, nodeClaims ...*v1.NodeClaim) error {
-	for _, nodeClaim := range nodeClaims {
-		// If we still get the NodeClaim, but it's already marked as terminating, we don't need to call Delete again
-		if nodeClaim.DeletionTimestamp.IsZero() {
-			if err := c.kubeClient.Delete(ctx, nodeClaim); err != nil {
-				return client.IgnoreNotFound(err)
-			}
-		}
+func (*Controller) associatedNodeClaimError(err error) error {
+	return reconcile.TerminalError(fmt.Errorf("failed to terminate node, expected a single associated nodeclaim, %w", err))
+}
+
+func (c *Controller) hasTerminationGracePeriodElapsed(nodeTerminationTime *time.Time) bool {
+	if nodeTerminationTime == nil {
+		return false
 	}
-	return nil
+	return !c.clock.Now().Before(*nodeTerminationTime)
```

Comment:
Can we just do `.After`? Yes, it's technically "not correct" but it practically doesn't matter and makes it so much more readable